php/libraries/NDB_BVL_InstrumentStatus_ControlPanel.class.inc

```diff
@@ -105,9 +105,13 @@ class NDB_BVL_InstrumentStatus_ControlPanel extends NDB_BVL_InstrumentStatus
 
         if (empty($this->subtest)) {
             $hasAccess = $this->_hasAccess();
-
-            if ($this->getDataEntryStatus() == null && $hasAccess) {
-                $this->setDataEntryStatus('In Progress');
+            $status    = new \NDB_BVL_InstrumentStatus;
```

Comment:
This looks like it's just peppering over a bigger problem.. why is `getDataEntryStatus() == null` not sufficient? Why do we need a `\NDB_BVL_InstrumentStatus` object that's separate from `$this->_status` to get the Data_entry property on it?

Reply:
@driusan getDataEntryStatus() returns null and even if there is a data entry status. It only returns the status when setDataEntryStatus has been called. I'm confused by it myself.

Reply:
please try just adding

```
        if ($this->_status === null && !empty($this->_commentID)) {
            $this->select($this->_commentID);
        }
```

at the top of getDataEntryStatus and see if that fixes it. It looks like that used to be done by NDB_Caller which doesn't exist anymore

Reply:
@driusan that works. See newest commit with it added.